src/btree/bt_sync.c

```diff
@@ -184,11 +199,30 @@ __sync_ref_obsolete_check(WT_SESSION_IMPL *session, WT_REF *ref)
             obsolete = __wt_txn_visible_all(session, newest_stop_txn, newest_stop_durable_ts);
         }
 
-        if (obsolete) {
+        if (obsolete && addr.type == WT_ADDR_LEAF_NO) {
             WT_REF_UNLOCK(ref, WT_REF_DELETED);
             WT_STAT_CONN_DATA_INCR(session, cc_pages_removed);
 
             WT_RET(__wt_page_parent_modify_set(session, ref, true));
+        } else if (obsolete && __sync_should_read_obsolete_page(session)) {
+            /*
+             * This page has overflow items, so we can't just drop it. Instead, read it into memory
+             * and mark it dirty (and ready to evict) so that the next reconciliation will clean it
+             * up. Only take this path for a limited number of pages so as not to churn the cache.
```

Comment:
I was all prepared to come back and comment on the throttle logic, then found that it was gone. That decision is fine with me. I'm happy to optimize if there is a need in the future.

Could you update this comment?